### PR TITLE
sales credit endpoint

### PR DESCRIPTION
**Pull request by charan41594**
## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[Y] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[Y] Tests
[ ] Other
```

## Checklist
Have you done the following?
```
[N] Exported any new types in index.ts
[Y] Merged master
[Y] Run `npm run lint`
[Y] Ensure the tests pass
```

### CODE REVIEW

File: src/BrightpearlClient.ts [hunk 1]
```diff
@@ -126,6 +127,25 @@ class BrightpearlClient {
...
+     */
+    public async getSalesCreditOrders(salesCreditOrderIds: number[], optionalParams?: OrderUrlParams[]): Promise<SalesCreditOrderInformation[]> {
+        const urlParams = optionalParams ? `?includeOptional=${optionalParams.join()}` : "";
+        const elementsToReplace: ElementsToReplace[] = [
+            {
+                key: ORDER_ID,
+                value: salesCreditOrderIds.join()
+            }
+        ];
+        const url = UrlHelper.getUrl(`${this.baseUrl}/${ENDPOINTS.order.get}`, elementsToReplace);
```
Comment (carlrc) [inline src/BrightpearlClient.ts]: Looks like this should be the sales credit endpoint not order GET

File: src/Endpoints.ts [hunk 2]
```diff
@@ -108,6 +108,10 @@ const ENDPOINTS  = Object.freeze({
             get: `${ORDER}/order-type`
         }
     },
+    sales_credit: {
+        base: `${ORDER}`,
```
Comment (carlrc) [inline src/Endpoints.ts]: Don't think this is necessary given you can't query the endpoint without a list of IDs http://api-docs.brightpearl.com/order/sales-credit/get.html

File: src/BrightpearlClient.ts [hunk 3]
```diff
@@ -126,6 +127,25 @@ class BrightpearlClient {
             {headers: this.getHeaders()});
         return httpResponse.data.response;
     }
+    /**
+     * Gets Sales Credit Orders
+     * @param salesCreditOrderIds - Sorted array of order ids
+     * @param optionalParams - Optionally pass in array of order url params
+     * @returns SalesCreditOrderInformation - Array of order information
+     */
+    public async getSalesCreditOrders(salesCreditOrderIds: number[], optionalParams?: OrderUrlParams[]): Promise<SalesCreditOrderInformation[]> {
```
Comment (carlrc) [inline src/BrightpearlClient.ts]: I can't see anything in the docs about http://api-docs.brightpearl.com/order/sales-credit/get.html accepting extra params like the ones that exist in the order url params?

File: src/IBrightpearlClient.ts [hunk 4]
```diff
@@ -1235,6 +1238,102 @@ interface PaginationParams {
...
 
+// BRIGHTPEARL SALES CREDIT ORDER INFORMATION
+interface SalesCreditOrderInformation {
+    id: number;
+    parentOrderId: number;
+    orderTypeCode: string;
+    reference: string;
+    version: number;
+    state: State;
+    orderStatus: OrderNote;
```
Comment (carlrc) [inline src/IBrightpearlClient.ts]: should orderStatus be of type OrderNote?

File: src/IBrightpearlClient.ts [hunk 5]
```diff
@@ -1235,6 +1238,102 @@ interface PaginationParams {
     firstResult?: number;
 }
 
+// BRIGHTPEARL SALES CREDIT ORDER INFORMATION
```
Comment (carlrc) [inline src/IBrightpearlClient.ts]: Not necessary

File: src/IBrightpearlClient.ts [hunk 6]
```diff
@@ -1235,6 +1238,102 @@ interface PaginationParams {
...
+interface TotalValue {
+    net: string;
+    taxAmount: string;
+    baseNet: string;
+    baseTaxAmount: string;
+    baseTotal: string;
+    total: string;
+}
+
+interface Parties {
```
Comment (carlrc) [inline src/IBrightpearlClient.ts]: Did this have to be re-created when it existed already?

File: src/IBrightpearlClient.ts [hunk 7]
```diff
@@ -1235,6 +1238,102 @@ interface PaginationParams {
...
+    quantity: Quantity;
+    itemCost: Cost ;
+    productPrice: Cost ;
+    discountPercentage: string;
+    rowValue: RowValue;
+    nominalCode: string;
+    composition: SalesCreditComposition;
+}
+
+interface RowValue {
```
Comment (carlrc) [inline src/IBrightpearlClient.ts]: Did this have to be re-created?

File: src/IBrightpearlClient.ts (hunk 4 above)
Comment (carlrc) [reply src/IBrightpearlClient.ts]: Could you provide an example of the sales credit response you worked off? The API documentation don't show a bunch of the fields that are included here.

File: src/IBrightpearlClient.ts [hunk 8]
```diff
@@ -1235,6 +1238,102 @@ interface PaginationParams {
...
+
+interface State {
+    tax: string;
+}
+
+interface Tax {
+    errors?: string[];
+}
+
+interface ShippingMethod {
```
Comment (carlrc) [inline src/IBrightpearlClient.ts]: There is already an interface named this

**Comment by carlrc** [comment]
New types should be exported

File: src/IBrightpearlClient.ts [hunk 9]
```diff
@@ -1235,6 +1238,102 @@ interface PaginationParams {
...
+interface RowValue {
+    taxRate?: string;
+    taxCode: string;
+    taxCalculator?: string;
+    rowNet: RowNet ;
+    rowTax: RowTax ;
+    taxClassId?: number;
+}
+
+interface SalesCreditComposition {
```
Comment (grahamcarpenter) [inline src/IBrightpearlClient.ts]: You should be able to just reuse the `OrderInformationComposition` interface

File: src/IBrightpearlClient.ts [hunk 10]
```diff
@@ -1235,6 +1238,102 @@ interface PaginationParams {
...
+
+interface Tax {
+    errors?: string[];
+}
+
+interface ShippingMethod {
+    shippingMethodId: number;
+}
+
+interface InvoicesEntity {
```
Comment (grahamcarpenter) [inline src/IBrightpearlClient.ts]: If this isn't used in the SalesCredit, please remove

File: src/IBrightpearlClient.ts [hunk 11]
```diff
@@ -1235,6 +1238,102 @@ interface PaginationParams {
...
+    acknowledged: number;
+    costPriceListId: number;
+    historicalOrder: boolean;
+}
+
+interface State {
+    tax: string;
+}
+
+interface Tax {
```
Comment (grahamcarpenter) [inline src/IBrightpearlClient.ts]: If this isn't used in the SalesCredit, please remove

File: src/IBrightpearlClient.ts [hunk 12]
```diff
@@ -1235,6 +1238,102 @@ interface PaginationParams {
...
+    assignment: Assignment;
+    parties: Parties;
+    orderRows: AttributeMap<OrderRowItem>;
+    warehouseId: number;
+    acknowledged: number;
+    costPriceListId: number;
+    historicalOrder: boolean;
+}
+
+interface State {
```
Comment (grahamcarpenter) [inline src/IBrightpearlClient.ts]: If this isn't used in the SalesCredit, please remove

File: src/IBrightpearlClient.ts (hunk 4 above)
Comment (grahamcarpenter) [reply src/IBrightpearlClient.ts]: ^ same here. I don't see many of the fields listed here in the docs for POST or GET.

**Comment by charan41594** [comment]
updated the endpoint, no changes required w.r.t interfaces.

File: src/BrightpearlClient.ts [hunk 13]
```diff
@@ -126,6 +126,23 @@ class BrightpearlClient {
             {headers: this.getHeaders()});
         return httpResponse.data.response;
     }
+    /**
+     * Gets Sales Credit Orders
+     * @param salesCreditOrderIds - Sorted array of order ids
+     * @returns SalesCreditInformation - Array of order information
+     */
+    public async getSalesCreditOrders(salesCreditOrderIds: number[]): Promise<OrderInformation[]> {
```
Comment (grahamcarpenter) [inline src/BrightpearlClient.ts]: Does the `SalesCredit GET` return the same fields as the `Order GET`?

Comment (charan41594) [reply src/BrightpearlClient.ts]: yes.

Comment (charan41594) [reply src/BrightpearlClient.ts]: actually it is same as `Sales Order GET`

File: src/BrightpearlClient.ts (hunk 1 above)
Comment (carlrc) [reply src/BrightpearlClient.ts]: ^^^

File: src/BrightpearlClient.ts [hunk 14]
```diff
@@ -126,6 +126,23 @@ class BrightpearlClient {
             {headers: this.getHeaders()});
         return httpResponse.data.response;
     }
+    /**
+     * Gets Sales Credit Orders
+     * @param salesCreditOrderIds - Sorted array of order ids
+     * @returns SalesCreditInformation - Array of order information
+     */
+    public async getSalesCreditOrders(salesCreditOrderIds: number[]): Promise<SalesOrderInformation[]> {
```
Comment (carlrc) [inline src/BrightpearlClient.ts]: If we are fetching an order, from the order GET endpoint, can't we just use the existing getSalesOrder method?

**Pull request merged**